### PR TITLE
Marota random redispatch agent

### PR DESCRIPTION
**Pull request by Tezirg**

### CODE REVIEW

**Comment by BDonnot** [comment]
I think this agent already exist. 
If you use the IdToVect converter and authorize only the redispatching, and call the "sample" method, this is exactly what it will do.
Or am I missing something ?

Beside using the IdToAct will authorize to take actions of different increment, that will never violate the ramp min / max. This is not the case with this agent (if the amount to redispatch is like 10 MW but the ramp is 5MW for a generator, you will have an ineffective agent).

File: grid2op/Agent/RandomRedispatchAgent.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        :redispatching_increment: `float`
+          The redispatching MW value used in both directions
+        """
+        super().__init__(action_space)
+        self.desired_actions = []
+
+        # Get all generators IDs
+        gens_ids = np.arange(self.action_space.n_gen, dtype=int)
+        # Filter out non resipatchable IDs
+        gens_redisp = gens_ids[self.action_space.gen_redispatchable == True]
```
Comment (BDonnot) [inline grid2op/Agent/RandomRedispatchAgent.py]: remove the "== True"
`gens_redisp = gens_ids[self.action_space.gen_redispatchable]`
evaluates faster and is more readable

**Review by BDonnot — CHANGES_REQUESTED** [review]
I don't know to what problem this class could solve better than the "IdToAct" class if you restrict it to only redispatching action (and maybe use filtering for the nb_max_generator is it's really needed).

The name is also completely misleading. It is not an Agent that takes at random uniformely amongs all possible redispatching actions. It takes some pre specified redispatching action at random...

However i found a clear setting in the case IdToAct does a much much better job (it always respects the ramps).
I you want to improve it by giving the number of "splits" you do for redispatch (instead of 7 both side, you can do 1 or 2 or 42)

I'm not found of the current implementation for these reasons.

**Comment by marota** [comment]
Le besoin est d'avoir un agent sur lequel on puisse faire des tests pour le redispatching facilement. 
Je n'ai pas vu ce type d'agent dans la liste des agents. Cet agent avait été revu avec Jean.
Si vous voyez comment faire différemment ou l'améliorer pas de soucis pour le changer

**Comment by BDonnot** [comment]
If this is for testing purpose I suggest we had a dedicated "test" or "testing" submodule to grid2op and put it there. Otherwise people will legitimately ask question about it trying to use it.
Beside, there are plenty of tests for redispatching already.

**Comment by BDonnot** [comment]
Will be added in the other PR (PR #97)

**Pull request closed without merging**